tests/test_transmissionCurve.py

```diff
@@ -0,0 +1,554 @@
+            tc2.getNaturalSampling().size
+        )
+        self.assertEqualOrNaN(tc.getThroughputAtBounds()[0], tc2.getThroughputAtBounds()[0])
+        self.assertEqualOrNaN(tc.getThroughputAtBounds()[1], tc2.getThroughputAtBounds()[1])
+        for point in points:
+            self.assertFloatsEqual(
+                tc.sampleAt(point, tc.getNaturalSampling()),
+                tc2.sampleAt(point, tc.getNaturalSampling())
+            )
+
+    def makeAndCheckConstant(self, curve, wavelengths, throughputAtMin, throughputAtMax):
+        """Construct a constant TransmissionCurve and apply basic tests to it."""
+        throughput = curve(wavelengths)
+        tc = lsst.afw.image.TransmissionCurve.makeConstant(throughput, wavelengths,
+                                                           throughputAtMin, throughputAtMax)
+        self.assertEqual(tc.getNaturalSampling().min, wavelengths[0])
+        self.assertEqual(tc.getNaturalSampling().max, wavelengths[-1])
+        self.assertFloatsEqual(
+            throughput,
+            tc.sampleAt(self.point, wavelengths)
+        )
+        self.assertFloatsEqual(
+            throughput,
+            useOutArg(tc.sampleAt, wavelengths.size, self.point, wavelengths)
+        )
+        if not np.isnan(throughputAtMin):
+            self.assertEqual(tc.getThroughputAtBounds()[0], throughputAtMin)
+            self.assertFloatsEqual(
+                throughputAtMin,
+                tc.sampleAt(self.point, np.array([2*wavelengths[0] - wavelengths[1]]))
+            )
+        else:
+            self.assertTrue(np.isnan(tc.getThroughputAtBounds()[0]))
+        if not np.isnan(throughputAtMax):
+            self.assertEqual(tc.getThroughputAtBounds()[1], throughputAtMax)
+            self.assertFloatsEqual(
+                throughputAtMax,
+                tc.sampleAt(self.point, np.array([2*wavelengths[-1] - wavelengths[-2]]))
+            )
+        else:
+            self.assertTrue(np.isnan(tc.getThroughputAtBounds()[1]))
+        # Check that we get decent results when interpolating to different wavelengths.
+        wavelengths1 = np.linspace(wavelengths[0] - 10, wavelengths[-1] + 10, 200)
+        if np.isnan(throughputAtMin):
+            if np.isnan(throughputAtMax):
+                mask = np.logical_and(wavelengths1 >= wavelengths[0], wavelengths1 <= wavelengths[-1])
+            else:
+                mask = wavelengths1 >= wavelengths[0]
+        else:
+            if np.isnan(throughputAtMax):
+                mask = wavelengths1 <= wavelengths[-1]
+            else:
+                mask = np.ones(wavelengths1.size, dtype=bool)
+        self.assertFloatsAlmostEqual(
+            curve(wavelengths1)[mask],
+            tc.sampleAt(self.point, wavelengths1)[mask],
+            rtol=1E-2, atol=1E-2
+        )
+        if mask.any():
+            self.assertTrue(
+                np.isnan(tc.sampleAt(self.point, wavelengths1)[np.logical_not(mask)]).all()
+            )
```

Comment:
This is actually testing that values that *should* be NaN are NaN (because they're beyond the bounds and that's how we set `throughputAtBounds`).  Will add a comment to that effect.

Reply:
I'm not sure how that addresses my question. What would go wrong if you tried to evaluate this assertion when `mask.any()` was false?